src/pipeline.jl

```diff
@@ -192,26 +192,41 @@ function _get_index(con::DB, source::String, on::Symbol)
     return getproperty(base, on)
 end
 
-function _set_tbl_col_impl(
-    con::DB,
-    source::String,
-    idx::Vector,
-    vals::Vector;
-    on::Symbol,
-    col::Symbol,
-    opts...,
-)
-    df = DF.DataFrame([idx, vals], [on, col])
-    tmp_tbl = "t_col_$(col)"
-    register_data_frame(con, df, tmp_tbl)
-    # FIXME: should be fill=error (currently not implemented)
-    res = create_tbl(con, source, tmp_tbl; on = [on], cols = [col], fill = false, opts...)
-    unregister_data_frame(con, tmp_tbl)
-    return res
+"""
+    as_table(op::Function, con::DB, name::String, args...)
+
+Temporarily "import" a Julia object into a DuckDB session.  It does it
+by first creating a `DataFrame`.  `args...` are passed on to the
+`DataFrame` constructor as is.  It is registered with the DuckDB
+connection `con` as the table `name`.  This function can be used with
+a `do`-block like this:
+
+    julia> as_table(con, "mytbl", (;col=collect(1:5))) do con, name
+               DD.query(con, "SELECT col, col+2 as 'shift_2' FROM '\$name'")
+           end |> DataFrame
+    5×2 DataFrame
+    Row │ col     shift_2
+         │ Int64?  Int64?
+    ─────┼─────────────────
+       1 │      1        3
+       2 │      2        4
+       3 │      3        5
+       4 │      4        6
+       5 │      5        7
```

Comment:
Could this be a doctest instead?

Reply:
I added it as doctest, however I'm not sure I did it correctly.  Particularly, I don't know if escaping the `$name` is correct.  Without it, tests fail because it's looking for a `name` variable in `pipeline.jl`.

Reply:
I think it's working, because the docs should be running doctests and they passed. Thanks for the update